console/v1/types_console_plugin.go

```diff
@@ -43,11 +43,138 @@ type ConsolePluginSpec struct {
 	Backend ConsolePluginBackend `json:"backend"`
 	// proxy is a list of proxies that describe various service type
 	// to which the plugin needs to connect to.
+	// +listType=map
+	// +listMapKey=alias
```

Comment:
Since this is in spec, and you don't otherwise have anything validating that the proxy aliases in the list are unique, I think this list will need to be marked atomic

If we were already validating uniqueness of the aliases, then this would be correct, but I don't see anything that prevents anyone having duplicate aliases right now